Updates to the Loading and Saving System

Spebby opened the pull request:
Instead of using the rudimentary and un-user friendly string loading/saving method, I've rewritten it to use the OS's built-in file browser to save and load assets.
<img width="1010" alt="image" src="https://user-images.githubusercontent.com/47935791/234734669-709c067d-490a-4e8d-8865-0839fcd9b7c8.png">
<img width="1023" alt="image" src="https://user-images.githubusercontent.com/47935791/234734732-e66f8279-2694-4d42-b628-7da1372da257.png">

Spebby committed:
Update StoryGraph.cs

Updated to allow for file explorer usage, instead of the current string based approach

Spebby committed:
Update GraphSaveUtility.cs

supports file browser

Spebby committed:
Merge remote-tracking branch 'origin/patch-2' into patch-1

Spebby committed:
Fixes :)

florian-rieder commented:
Looks good ! Although, do you not encounter a breaking bug when loading a narrative, where most edges would disappear and most nodes are moved to 0,0 (I assume) ? See #10. I'll keep looking into it.

Spebby commented:
> Looks good ! Although, do you not encounter a breaking bug when loading a narrative, where most edges would disappear and most nodes are moved to 0,0 (I assume) ? See #10. I'll keep looking into it.

no  i don't. what version are you testing?

Spebby commented:
> Regarding your PR, I feel like it's kinda tiring to have the Resources folder pop up in the native file explorer when saving every time.

I thought I had fixed this in this Patch, turns out it's only on my personal fork. The approach I did on that branch was to only have the OS's explorer open when you save a new file, but repeat saves of the same file wouldn't cause this to happen. Will quickly implement that patch, then I'll look into the File Name bug and renaming the buttons later today.

Spebby committed:
Reordered some code

Moved the Reveal in Explorer call to be inside the pathless save function, since the only time you'd actually care where the file saved is when you first save it.

Spebby committed:
addressed some suggestions

florian-rieder commented:
LGTM
